addons/account/tests/test_reconciliation.py

```diff
@@ -1444,3 +1445,341 @@ def test_reconciliation_cash_basis02(self):
                 expected['tax_10']
             )
             index += 1
+
+    def test_reconciliation_cash_basis_fx_01(self):
+        """
+        Having issued an invoice at date Nov-21-2018 as:
+
+        Expenses            5,301.00 USD         106,841.65              0.00
+        Taxes                 848.16 USD          17,094.66              0.00
+            Payables       -6,149.16 USD               0.00        123,936.31
+
+        On Dec-20-2018 user issues an FX Journal Entry as:
+
+        Payables                0.00 USD             167.86             0.00
+            FX Gains            0.00 USD               0.00           167.86
+
+        On Same day user records a payment for:
+
+        Payables            6,149.16 USD         123,768.45              0.00
+            Bank           -6,149.16 USD               0.00        123,768.45
+
+        And then reconciles the Payables Items which shall render only one Tax
+        Cash Basis Journal Entry because of the actual payment, i.e.
+        amount_currency != 0:
+
+        Tax Base Acc.           0.00 USD         106,696.94              0.00
+            Tax Base Acc.       0.00 USD               0.00        106,696.94
+        Creditable Taxes      848.16 USD          17,071.51              0.00
+            Taxes            -848.16 USD               0.00         17,071.51
+        """
+
+        company = self.env.ref('base.main_company')
+        company.tax_cash_basis_journal_id = self.cash_basis_journal
+
+        aml_obj = self.env['account.move.line'].with_context(
+            check_move_validity=False)
+
+        # Purchase
+        purchase_move = self.env['account.move'].create({
+            'name': 'purchase',
+            'journal_id': self.purchase_journal.id,
+        })
+
+        aml_obj.create({
+            'name': 'expenseTaxed',
+            'account_id': self.expense_account.id,
+            'debit': 106_841.65,
+            'move_id': purchase_move.id,
+            'tax_ids': [(4, self.tax_cash_basis.id, False)],
+            'currency_id': self.currency_usd_id,
+            'amount_currency': 5_301.00,
+        })
+        aml_obj.create({
+            'name': 'TaxLine',
+            'account_id': self.tax_waiting_account.id,
+            'debit': 17_094.66,
+            'move_id': purchase_move.id,
+            'tax_line_id': self.tax_cash_basis.id,
+            'currency_id': self.currency_usd_id,
+            'amount_currency': 848.16,
+        })
+        purchase_payable_line0 = aml_obj.create({
+            'name': 'Payable',
+            'account_id': self.account_rsa.id,
+            'credit': 123_936.31,
+            'move_id': purchase_move.id,
+            'currency_id': self.currency_usd_id,
+            'amount_currency': -6_149.16,
```

Comment:
python 3.6 not supported

Reply:
nope, it is the opposite. Odoo does need python >= 3.5, hence, every piece of code should comply to py3.5, not above

Reply:
Got it!